test/e2e/storage/drivers/in_tree.go

```diff
@@ -260,11 +258,16 @@ func (g *glusterFSDriver) GetDriverInfo() *testsuites.DriverInfo {
 	return &g.driverInfo
 }
 
-func (g *glusterFSDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
-	framework.SkipUnlessNodeOSDistroIs("gci", "ubuntu", "custom")
+func (g *glusterFSDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) bool {
+	if !framework.NodeOSDistroIs("gci", "ubuntu", "custom") {
+		return true
+	}
 	if pattern.FsType == "xfs" {
-		framework.SkipUnlessNodeOSDistroIs("ubuntu", "custom")
+		if !framework.NodeOSDistroIs("ubuntu", "custom") {
+			return true
+		}
 	}
+	return false
```

Comment:
nit: 
```golang
!framework.NodeOSDistroIs("gci", "ubuntu", "custom") || 
(pattern.FsType == "xfs" && 
!framework.NodeOSDistroIs("ubuntu", "custom"))
```

(negated if you are changing the function name)

Reply:
Yep.